Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift

```diff
@@ -360,3 +358,26 @@ extension PlaceCarouselViewController: LocationProvider {
     }
 }
 
+extension PlaceCarouselViewController: PlacesControllerDelegate {
+    func placesControllerNoPlacesFound(_ controller: PlacesController, willRetry retry: Bool) {
+        if retry {
+            // TODO placeholder for the waiting state.
+            headerView.numberOfPlacesLabel.text = "Waiting"
+        } else {
+            self.places = []
+        }
+
+    }
+
+    func placesController(_ controller: PlacesController, didUpdatePlaces places: [Place]) {
+        self.places = places
```

Comment:
can we please have the full functionality from the removed closure on the old `getPlaces` request?
```
if places != self.places {
    self.places = places
    self.placeCarousel.refresh()
}
```